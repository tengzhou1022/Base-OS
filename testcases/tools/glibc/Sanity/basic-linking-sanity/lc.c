#include <stdio.h>
#include <stdlib.h>
#include <string.h>

int main(){
  char one[] = "Whatever!";
  char *two;
  FILE *fp = fopen("lc.out", "w");

  fprintf(fp, "Printf: %d\n", 10);
  fprintf(fp, "Whatever: %s, length %d\n", one, strlen(one));

  two = malloc(sizeof(char) * 20);
  memcpy(two, one, 3);
  two[3] = '\0';
  fprintf(fp, "Two: %s, length %d\n", two, strlen(two));

  close(fp);
  free(two);
  return 0;
}
