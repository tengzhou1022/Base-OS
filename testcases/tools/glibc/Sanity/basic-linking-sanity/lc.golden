Printf: 10
Whatever: Whatever!, length 9
Two: Wha, length 3
